docs/docs/contribute/docs/write-documentation.md

```diff
@@ -0,0 +1,270 @@
+
+The root directory for our site is located at `/docs/docs`, with `index.md` being the file for our homepage. Subsequent directories are placed within folders.
+
+![A level in the doc's hierarchy.](assets/docs-hierarchy.png)
+
+Each level contains:
+
+- An `index.md` file as the landing page
+
+- An (optional) `assets` folder to hold media used in that level
+
+- A `.pages` file to organize the order of our pages
+
+    ![The .pages file helps to structure the hierarchy of each level.](assets/editing-ref-pages.png)
+
+    !!! note
+        If `...` is used within `.pages`, files in the same hierarchy that have not been mentioned in `.pages` will be arranged in alphabetical order.
+
+        ![Using ... to automatically list pages.](assets/editing-ref-pages-dots.png)
+
+## Generating Documentation for the Shards API
+
+You might have observed that the `/docs/docs/reference/shards` folder is conspicuously empty. This is to allow for the faster previewing of your static site without having to load all the pages for the Shards API.
+
+To generate the documentation for the Shards API:
+
+1. Launch your MinGW terminal.
+
+2. Navigate to where your Shards repository is located using the command `cd $(cygpath -u '(X)')`, where (X) is the directory of your folder.
+
+        cd $(cygpath -u '(X)')
+
+    If your repository is located at `C:\Fragcolor\Shards`, the command used in the MinGW terminal would be `cd $(cygpath -u 'C:\Fragcolor\Shards')`.
+
+ 3. Input the following command:
+
+        ./build/Debug/shards ./docs/generate.edn
+
+!!! note
+    The **Debug** version of Shards must be used for the generating of the Shards API pages.
+
+Once done, you will notice that the directory is now filled with folders and Markdown files of various shards.
+
+## Generating Log Files for Examples
+
+Did you realize that the "Output" segments for your code examples are empty too? This is due to how a `.log` file is required for each example to show its output.
+
+To generate the `.log` files for your code examples:
+
+1. Launch your MinGW terminal.
+
+2. Navigate to where your Shards repository is located using the command `cd $(cygpath -u '(X)')`, where (X) is the directory of your folder.
+
+        cd $(cygpath -u '(X)')
+
+    If your repository is located at `C:\Fragcolor\Shards`, the command used in the MinGW terminal would be `cd $(cygpath -u 'C:\Fragcolor\Shards')`.
+
+ 3. Input the following command:
+
+        ./run-samples.sh --release
+
+!!! warning "Segmentation fault"
+    If the error for "Segmentation fault" appears, run the command below before attempting Step 3 again.
+
+        export GFX_BACKEND=D3D12
+
+When the script is done, you will notice new `.log` files within your "Samples" folder. These will be shown in your documentation as the "Output" of your code examples.
+
+![The results with the .edn.log file.](assets/editing-examples-log-md.png)
+
+![The log file will populate the "Output" segment.](assets/editing-examples-site-output.png)
```

Comment:
I will leave it as it is first since the verbose version is not something we want to showcase. It is also mostly for example purposes. If anything, a fictional example would work too rather than using a non-ideal example.